Makefile

```
@@ -230,8 +230,22 @@ test: test-shell ## Runs unit tests with race detector and coverage (use -short
+.PHONY: check-coverage-threshold
+check-coverage-threshold:
+	@if [ -z "$(COVERAGE_THRESHOLD)" ] || [ "$(COVERAGE_THRESHOLD)" = "null" ]; then \
+		echo "ERROR: coverage threshold unavailable from .settings.yaml (quality.coverage_threshold)"; \
+		echo "       Check that yq is installed ('make tools-setup') and that"; \
+		echo "       .settings.yaml defines quality.coverage_threshold."; \
+		exit 1; \
+	fi
+
 .PHONY: test-coverage
-test-coverage: test ## Runs tests and enforces coverage threshold (from .settings.yaml quality.coverage_threshold)
+test-coverage: check-coverage-threshold test ## Runs tests and enforces coverage threshold (from .settings.yaml quality.coverage_threshold)
```

checkmake (0.3.2)

[warning] Target body for "test-coverage" exceeds allowed length of 5 lines (7).

(maxbodylength)